src/function/memo.rs

```diff
@@ -119,23 +135,46 @@ pub(super) struct Memo<V> {
     /// as the current revision.
     pub(super) verified_at: AtomicCell<Revision>,
 
+    /// Is this memo verified to not be a provisional cycle result?
+    pub(super) verified_final: AtomicBool,
+
     /// Revision information
     pub(super) revisions: QueryRevisions,
 }
 
 // Memo's are stored a lot, make sure their size is doesn't randomly increase.
 // #[cfg(test)]
 const _: [(); std::mem::size_of::<Memo<std::num::NonZeroUsize>>()] =
-    [(); std::mem::size_of::<[usize; 12]>()];
+    [(); std::mem::size_of::<[usize; 17]>()];
```

Comment:
That's quite the regression, boxing the cycle heads as said in another review should bring this down to 14 I believe at least. Then the two usizes mainly come from the two fields being added.

Reply:
Done